plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/server/debug/ReplicationControllerListenerJob.java

```diff
@@ -66,7 +66,7 @@ public void connectionChanged(IConnection connection, String property, Object ol
 				String podName = candidate.getName();
 				if (!oldPods.contains(podName) &&
 					"Running".equals(candidate.getStatus()) &&
-					ResourceUtils.containsAll(dcSelector, candidate.getLabels())
+					ResourceUtils.containsAll(selector, candidate.getLabels())
```

Comment:
to improve code readability: we could extract this logic (l64-l71), that basically checks if the given pod is a new, running pod to a method #isNewControlledPod and store the pod if the method returns true.